### PR TITLE
Add secret filters and improve UI

### PR DESCRIPTION
**Pull request by agix**
You can now see your secret or only the shared ones.
Account creation need password confirmation.
Add secret and Add folder are more visible.

### CODE REVIEW

File: src/components/Sidebar.js [hunk 1]
```diff
@@ -36,15 +37,29 @@ function Sidebar() {
         <ul className="sidebar-menu">
           <SidebarMenuLink to="/secrets/" activeOnlyWhenExact>
             <Icon id="home" size="base" />
-            Home
+            {secretin.currentUser.username}
```
Comment (Calyhre) [inline src/components/Sidebar.js]: Current user must be in a store.

File: src/components/Sidebar.js [hunk 2]
```diff
@@ -36,15 +37,29 @@ function Sidebar() {
...
           </SidebarMenuLink>
+          <div className="sidebar-separator">
+            <div />
```
Comment (Calyhre) [inline src/components/Sidebar.js]: you don't need this empty div. Just `<div className="sidebar-separator" />`

File: src/components/Sidebar.js [hunk 3]
```diff
@@ -36,15 +37,29 @@ function Sidebar() {
...
+          <SidebarMenuLink to="/secrets/mine" activeOnlyWhenExact>
+            <Icon id="user" size="base" />
+            My secrets
+          </SidebarMenuLink>
+          <SidebarMenuLink to="/secrets/shared" activeOnlyWhenExact>
+            <Icon id="people" size="base" />
+            Shared secrets
+          </SidebarMenuLink>
+          <div className="sidebar-separator">
+            <div />
```
Comment (Calyhre) [inline src/components/Sidebar.js]: Same here, useless empty div

File: src/components/secrets/SecretList.js [hunk 4]
```diff
@@ -52,9 +60,31 @@ class SecretList extends Component {
...
+    const special = this.props.showAll || this.props.showMine || this.props.showShared;
+
+    let secrets = this.props.secrets
+      .filter(secret => fuzzyRegexp.test(secret.title));
+
+    let folders = new Immutable.Map();
+
+    if (special) {
+      secrets.forEach((secret) => {
+        const foldersSeq = secret.getIn(['users', secretin.currentUser.username, 'folders']).entrySeq();
```
Comment (Calyhre) [inline src/components/secrets/SecretList.js]: Please avoid using secretin in components. The cuttentUser is already in `AppUIStore`

File: src/components/secrets/SecretList.js [hunk 5]
```diff
@@ -52,9 +60,31 @@ class SecretList extends Component {
...
+        foldersSeq.forEach((folder) => {
+          folders = folders.setIn([folder[0], 'secrets', secret.id], secret);
+          if (folder[0] === 'ROOT') {
+            folders = folders.setIn([folder[0], 'name'], '');
+            folders = folders.setIn([folder[0], 'root'], true);
+          } else {
+            folders = folders.setIn([folder[0], 'name'], folder[1].get('name'));
+          }
+        });
+      });
```
Comment (Calyhre) [inline src/components/secrets/SecretList.js]: Two forEach to mutate a var? There must be a way better way to do it.

File: src/components/secrets/SecretList.js [hunk 6]
```diff
@@ -93,47 +130,82 @@ class SecretList extends Component {
...
-        </div>
       </div>
     );
   }
 
   render() {
+    let icon;
+    let title;
+    let link;
+    let special = false;
```
Comment (Calyhre) [inline src/components/secrets/SecretList.js]: What does special is? Please name it correctly

File: src/components/secrets/SecretListBreadcrumb.js [hunk 7]
```diff
@@ -67,7 +68,7 @@ function SecretListBreadcrumb({ folders }) {
             />
           )
           .unshift(
-            <Title key="home" title="Home" icon="home" link="/secrets/" />
+            <Title key="home" title={secretin.currentUser.username} icon="home" link="/secrets/" />
```
Comment (Calyhre) [inline src/components/secrets/SecretListBreadcrumb.js]: Same, remove secretin dependency in the component

File: src/components/secrets/SecretListItem/Options.js [hunk 8]
```diff
@@ -14,42 +14,56 @@ const propTypes = {
...
+        {
+          secret.type !== 'folder' &&
+            <Dropdown.MenuItem
+              onSelect={() => ShowSecretUIActions.showSecret({ secret, tab: secret.type === 'folder' ? 'access' : 'details' })}
+            >
+              Show
+            </Dropdown.MenuItem>
+        }
+        {
+          canShare &&
```
Comment (Calyhre) [inline src/components/secrets/SecretListItem/Options.js]: I would have let them disabled.

File: src/components/secrets/SecretListItem/Secret.js [hunk 9]
```diff
@@ -14,7 +14,7 @@ const propTypes = {
   secret: PropTypes.any,
   parentFolderId: PropTypes.string,
   isDragging: PropTypes.bool,
-  connectDragSource: PropTypes.func.isRequired,
+  connectDragSource: PropTypes.func.isRequired
```
Comment (Calyhre) [inline src/components/secrets/SecretListItem/Secret.js]: Stupid eslint, we need to fix it.

File: src/components/utilities/Input.js [hunk 10]
```diff
@@ -124,14 +124,14 @@ class Input extends Component {
         />
         {
           this.props.type === 'password' && (
-            <div className={this.state.showPassword ? 'eye-show' : 'eye-hide'}>
+            <div className="eye">
```
Comment (Calyhre) [inline src/components/utilities/Input.js]: Please use a more explicit classname here. `input--password-show` for example.

File: src/stylesheets/components/secrets/SecretListItem.sass [hunk 11]
```diff
@@ -57,3 +59,12 @@
...
   opacity: 1
+
+.secret-list-folder
+  height: 25px
+  background-color: rgba($color-brand, .02)
+  text-align: center
+  td
+    padding: 0 0 0 10px
+    border-bottom: 1px solid rgba($color-brand, .05)
+    vertical-align: middle
```
Comment (Calyhre) [inline src/stylesheets/components/secrets/SecretListItem.sass]: Please end your files with a new line.

File: src/stylesheets/components/users/UserConnect.sass [hunk 12]
```diff
@@ -19,3 +20,9 @@
   text-align: center
   margin: 10px 0 0
   color: $color-peter-river
+
+.user-connect-create
+  text-align: center
+  margin: 10px 0 0
+  a
+    color: $color-peter-river
```
Comment (Calyhre) [inline src/stylesheets/components/users/UserConnect.sass]: Same comment about the new line

File: src/stylesheets/components/utilities/Input.sass [hunk 13]
```diff
@@ -90,14 +90,8 @@
...
   float: right
   margin-top: -44px
   margin-right: 10px
-
-.eye-show
-  opacity: 1
-
-.eye-hide
-  opacity: 0.1
+  opacity: 0.7
```
Comment (Calyhre) [inline src/stylesheets/components/utilities/Input.sass]: Same comment abut the new line

File: src/stylesheets/components/utilities/Page.sass [hunk 14]
```diff
@@ -18,16 +18,25 @@
...
+  margin-top: -10px
+
+  .secret-list-new
+    display: inherit
+    margin-right: 10px
+    margin-top: 3px
+    .button
+      margin-left: 10px
+    .button:first-child
+      margin-left: 0px
```
Comment (Calyhre) [inline src/stylesheets/components/utilities/Page.sass]: Same comment abut the new line

File: src/components/Sidebar.js (hunk 2 above)
Comment (agix) [inline src/components/Sidebar.js]: It didn't work without empty div

File: src/components/secrets/SecretList.js (hunk 4 above)
Comment (agix) [inline src/components/secrets/SecretList.js]: Ok, I didn't check where I could find it. I will change it everywhere bro :).

File: src/components/Sidebar.js (hunk 2 above)
Comment (Calyhre) [inline src/components/Sidebar.js]: Then you are doing it wrong :)

File: src/components/secrets/SecretList.js (hunk 5 above)
Comment (agix) [inline src/components/secrets/SecretList.js]: I tried hard with a lot of Immutable function but didn't succeed. Maybe groupBy

File: src/components/Sidebar.js (hunk 2 above)
Comment (agix) [inline src/components/Sidebar.js]: Do it right ! It's CSS shit :D

File: src/components/secrets/SecretList.js (hunk 6 above)
Comment (agix) [inline src/components/secrets/SecretList.js]: mofo

File: src/components/secrets/SecretListItem/Options.js (hunk 8 above)
Comment (agix) [inline src/components/secrets/SecretListItem/Options.js]: Big debate here

**Pull request merged**